src/Compiler/TypedTree/CompilerGlobalState.fs

```diff
@@ -19,14 +17,17 @@ open FSharp.Compiler.Text
 /// policy to make all globally-allocated objects concurrency safe in case future versions of the compiler
 /// are used to host multiple concurrent instances of compilation.
 type NiceNameGenerator() =    
-    let basicNameCounts = ConcurrentDictionary<string,Ref<int>>(max Environment.ProcessorCount 1, 127)
+    let basicNameCounts = ConcurrentDictionary<string, int>(max Environment.ProcessorCount 1, 127)
+    // Cache this as a delegate.
+    let basicNameCountsUpdateDelegate = Func<string, int, int>(fun _k count -> count + 1)
```

Comment:
This should keep things thread-safe here, right?

Reply:
Yes. This use-case is actually used in the example in the docs for this method :)

https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.addorupdate?view=net-8.0#examples

Reply:
I ended up going back to `GetOrAdd` in 02546c5 anyway.